ADR/0008-integration-service-api-contracts.md

```diff
@@ -0,0 +1,248 @@
+
+Tasks from the test stream may exchange information through
+
+1. Parameters passed at the time of the trigger of a PipelineRun (that is user, git repository, etc)
+2. Output of upstream elements of the Pipeline which may be needed for performing the Task (that is build artifacts, container image, etc)
+3. Providing whether the Task was successful or not
+4. Output of the Task that may be needed for processing subsequent Tasks of the pipeline
+5. Output of the Task that need to be stored for auditing and troubleshooting (that is summarized results of a validation or a scan, etc)
+6. Communication with external systems that are needed for performing the Task (that is container image repository, image scanner, etc)
+7. Connection details and credentials that may be required for the above
+
+Related to [PLNSRVCE-41](https://issues.redhat.com/browse/PLNSRVCE-41) investigations in regards to [Tekton results](https://github.com/tektoncd/results). The investigations are documented [here](https://docs.google.com/document/d/1WGWx6MllVs-BwRLd0PP5QTQHgGp0fW2dOYRKRAKWiwo/edit#heading=h.bpyxfhyud4dv).
+
+## Decision
+
+### Tekton Task Results
+
+The output of each Tekton task will be provided in a minimized [Tekton result](https://tekton.dev/docs/pipelines/tasks/#emitting-results) in JSON format listing all test failures. The name of the result will be **HACBS_TEST_OUTPUT**
+
+The maximum size of a [Task's Results](https://tekton.dev/vault/pipelines-v0.17.3/tasks/#emitting-results) is limited by the container [termination message](https://kubernetes.io/docs/tasks/debug/debug-application/determine-reason-pod-failure/#customizing-the-termination-message) feature of Kubernetes.
+
+App Studio builds are structured as [a shared Persistent Volume per StoneSoup Workspace](https://docs.google.com/document/d/1IPlihVjkJ4Kb9tdhsk7iz3bn5rkT_SvJCNQhyXzK3aI/edit#bookmark=id.gefgys3vno2). This allows teams to share builds, implement caching and other shared volumes. A single persistent volume is mapped to each default build pipeline. Builds are passed a directory specific to their builds.
+
+#### Tekton Result Format
+
+The Test output of the Tekton result will be a JSON object that includes the context about the test along with the list of check names for all failed checks
+
+The output will provide the following information about the overall test result:
+- **result** - The outcome of the testing task, can be `SUCCESS`, `FAILURE`, `WARNING`, `SKIPPED` or `ERROR`
+- **namespace** - Optional, the rego namespace of the test policy
+- **timestamp** - An UNIX epoch timestamp of the test completion time
+- **successes** - The number of successful checks in the form of an integer
+- **note** - Optional, a short note providing additional information about the test
+- **failures** - The number of failed checks in the form of an integer
+- **warnings** - The number of warning checks in the form of an integer
```

Comment:
**successes**, **failures**, and **warnings** are implied non-optional but the examples below do not all include these fields. Are these required or optional?

Reply:
Thanks, I think we use `required` because they are required for specific status, such as  `failures` is required for the status is `FAILURE`. But it may be better to list all fields of `successes`, `failures`, and `warnings` for all result `SUCCESS`, `FAILURE`, `WARNING`, `SKIPPED` or `ERROR`.  Is it? @dirgim

Reply:
Yes, I agree that we can have those fields as required to show up for all statuses.